docs/DeveloperGuide.md

```diff
@@ -381,9 +383,62 @@ The _sequence diagram_ below shows how the components interact with each other w
 
 ![Sequence diagram of sort command](images/SortSequenceDiagram.png)
 
-Note that
+### Home Page UI
+
+##### Overview
+
+`HomeCard` is a new feature in the UI component designed to provide users with a quick overview of crucial information immediately upon application launch. It serves as a dynamic dashboard, displaying the current date and time, the total number of contacts, a summary of financial transactions, and a list of contacts available for the day.
+
+##### Implementation
+
+The `HomeCard` is implemented using JavaFX and is integrated into the main window of the application. It interacts directly with the `Model` component to fetch real-time updates, ensuring that all displayed information reflects the latest data.
+
+Key features include:
+
+- **Real-Time Clock**: Utilizing Java's `LocalDateTime` and `Timeline` to update every second.
+- **Contact Counter**: Dynamically updates as contacts are added or removed.
+- **Financial Overview**: Uses a `BarChart` to graphically display money owed and owing.
+- **Availability List**: Shows contacts available today based on their schedules.
+
+#### Detailed Design
+
+The layout for `HomeCard` is defined in `HomeCard.fxml`, organized to provide immediate access to its features, which are essential for enhancing user interaction and providing a comprehensive view at a glance.
+
+![Sequence diagram of Home Page UI](images/HomeCardSequenceDiagram.png)
+
+### MiniPersonCard component
+
+#### Overview
+
+Embedded within the `HomeCard`, the `MiniPersonCard` serves as a compact display module for contacts available on the current day. It offers a quick snapshot of essential contact details, enhancing the `HomeCard` functionality by allowing users to identify key information swiftly.
+
```

Comment:
Good to screenshot and label which component you're talking about. Likewise for DisplayCard